include/sacn/receiver.h

```diff
@@ -93,13 +92,32 @@ typedef struct SacnRemoteSource
   char name[SACN_SOURCE_NAME_MAX_LEN];
 } SacnRemoteSource;
 
+/*! Information about a sACN source that was found. */
+typedef struct SacnFoundSource
+{
+  /*! The Source Description. */
+  SacnRemoteSource info;
+  /*! The address from which we received these initial packets. */
+  EtcPalSockAddr from_addr;
+  /*! The per-universe priority. */
+  uint8_t priority;
+  /*! The DMX (startcode 0) data. */
+  uint8_t values[DMX_ADDRESS_COUNT];
```

Comment:
I would make this and the per_address data `const uint8_t*` for now. This follows the data ownership paradigm we have in the RDMnet library, where data ownership is never transferred through API calls or callbacks. It potentially allows the implementation to reuse buffer space.

Reply:
I disagree, but am open to discussion.  This is not a simple pass through.  The library must buffer the latest data and per-address packet for a source from the received socket until it determines it has a stable source set to notify the application.  The application would never own this memory.  This makes it more clear than passing a const uint8_t * to some internal-only data structure.

Reply:
I guess I'm thinking about how the RDMnet library is implemented - received data is copied into the stream buffer and parsed, with lots of memory usage optimizations. In a lot of cases, the library passes back a pointer to data that is still in the same buffer that it was copied into by the recv() call. These public structs are only created and assigned to when a callback is about to be delivered - they are different than the internal state representations that the library uses. That's why I wanted to use all pointers, to save copying data.

I think const uint8_t* and const uint8_t[] imply the same thing about ownership (that the data is still owned by the library), but I was just thinking that const uint8_t* would give you more implementation flexibility.

Reply:
Oh -- I think I understand now.  It's not the fact that the library owns the buffers in this case, it's the fact that the structure holds const uint8_t[] vs. const uint8_t*?  

I thought you were arguing something deeper in regards to ownership. In general, I totally agree with the memory ownership philosophy that you stated in your last comment, but I thought you were trying to apply it here -- which meant that we would have to ask for memory from the application for a source it hasn't even seen yet..

Based on your answer, I'll either make a change tomorrow or we'll talk more. :)

Reply:
Sorry for not being clear, when I said that ownership is never _transferred_ through API calls or callbacks, I didn't mean that meant the library user always owned the memory.

In this case, the library owns and retains ownership of the memory. Ownership is not transferred.

Yes, my comment was to change the array to a pointer. The pointer would still be to memory held within the library and the library would retain ownership.